File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& v) {
        int n = v.size();
        
        for (int i = n - 1; i >= 0; i--) {
            v[i]++;  // Add 1 to the current digit
            
            if (v[i] < 10) {
                return v;  // No carry
            }
            
            v[i] = 0;  // Carry occurred set digit to 0
        }
        
        // Edge case If all digits were 9 
        v.insert(v.begin(), 1);
        return v;
    }
};

File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        if (digits == null || digits.length == 0) {
            return new int[]{1};
        }
        int n = digits.length;
        for(int i = n - 1; i >= 0; i--){
            digits[i]++;

            if(digits[i] < 10) return digits;

            digits[i] = 0;
        }

        int []result = new int[n + 1];
        result[0] = 1;
        return result;
    }
}

